src/child_process_tracker.rs

```diff
@@ -0,0 +1,160 @@
+use anyhow::Result;
+use std::sync::Arc;
+
+use tokio::process::Child;
+
+/// Windows does not have a concept of parent processes and so
+/// killing the deno task process will not also kill any spawned
+/// processes by default. To make this work, we can use winapi's
+/// jobs api, which allows for associating a main process so that
+/// when the main process terminates, it will also terminate the
+/// associated processes.
+///
+/// Read more: https://stackoverflow.com/questions/3342941/kill-child-process-when-parent-process-is-killed
+#[derive(Clone)]
+pub struct ChildProcessTracker(Arc<dyn Tracker>);
+
+impl ChildProcessTracker {
+  #[cfg(windows)]
+  pub fn new() -> Self {
+    match windows::WinChildProcessTracker::new() {
+      Ok(tracker) => Self(Arc::new(tracker)),
+      Err(err) => {
+        if cfg!(debug_assertions) {
+          panic!("Could not start tracking processes. {:#}", err);
+        } else {
+          // fallback to not tracking processes if this fails
+          Self(Arc::new(NullChildProcessTracker))
+        }
+      }
+    }
+  }
+
+  #[cfg(not(windows))]
+  pub fn new() -> Self {
+    Self(Arc::new(NullChildProcessTracker))
+  }
+
+  pub fn track(&self, child: &Child) {
+    if let Err(err) = self.0.track(child) {
+      if cfg!(debug_assertions) && child.id().is_some() {
+        panic!("Could not track process: {:#}", err);
+      }
+    }
+  }
+}
+
+trait Tracker: Send + Sync {
+  fn track(&self, child: &Child) -> Result<()>;
+}
+
+struct NullChildProcessTracker;
+
+impl Tracker for NullChildProcessTracker {
+  fn track(&self, _: &Child) -> Result<()> {
+    Ok(())
+  }
+}
+
+#[cfg(target_os = "windows")]
+mod windows {
+  use anyhow::bail;
+  use anyhow::Result;
+  use std::ptr;
+  use tokio::process::Child;
+  use winapi::shared::minwindef::DWORD;
+  use winapi::shared::minwindef::LPVOID;
+  use winapi::shared::minwindef::TRUE;
+  use winapi::um::handleapi::INVALID_HANDLE_VALUE;
+  use winapi::um::jobapi2::AssignProcessToJobObject;
+  use winapi::um::jobapi2::CreateJobObjectW;
+  use winapi::um::jobapi2::SetInformationJobObject;
+  use winapi::um::winnt::JobObjectExtendedLimitInformation;
+  use winapi::um::winnt::HANDLE;
+  use winapi::um::winnt::JOBOBJECT_EXTENDED_LIMIT_INFORMATION;
+  use winapi::um::winnt::JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+
+  use super::Tracker;
+
+  pub struct WinChildProcessTracker {
+    handle: WinHandle,
+  }
+
+  impl WinChildProcessTracker {
+    pub fn new() -> Result<Self> {
+      unsafe {
+        let handle = CreateJobObjectW(ptr::null_mut(), ptr::null());
+        let mut info: JOBOBJECT_EXTENDED_LIMIT_INFORMATION = std::mem::zeroed();
+        info.BasicLimitInformation.LimitFlags =
+          JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
```

Comment:
I would suggest to add `JOB_OBJECT_LIMIT_BREAKAWAY_OK 
 | JOB_OBJECT_LIMIT_SILENT_BREAKAWAY_OK`.

This allows subprocesses to spawn other programs that run in the background.

Reply:
+ `JOB_OBJECT_LIMIT_DIE_ON_UNHANDLED_EXCEPTION`

Reply:
Thanks! Added all three.